src/t8_vtk/t8_vtk_reader.cxx

```diff
@@ -298,13 +302,108 @@ t8_vtk_iterate_cells (vtkSmartPointer < vtkDataSet > vtkGrid,
+  t8_gloidx_t         last_tree;
+  if (mpirank == main_proc) {
+    first_tree = 0;
+    last_tree = num_trees - 1;
+  }
+  /* Communicate the dimension to all processes */
+  sc_MPI_Bcast (&dim, 1, sc_MPI_INT, main_proc, comm);
+  t8_cmesh_set_dimension (cmesh, dim);
+  /* Communicate the number of trees to all processes. 
+   * TODO: This probably crashes when a vtkGrid is distributed in many 
+   * files. */
+  sc_MPI_Bcast (&num_trees, 1, T8_MPI_GLOIDX, main_proc, comm);
+
+  /* Build the partition. */
+  if (mpirank < main_proc) {
+    first_tree = 0;
+    last_tree = -1;
+    t8_geometry_c      *linear_geom = t8_geometry_linear_new (dim);
+    t8_cmesh_register_geometry (cmesh, linear_geom);
+  }
+  else if (mpirank > main_proc) {
+    first_tree = num_trees;
+    last_tree = num_trees - 1;
+    t8_geometry_c      *linear_geom = t8_geometry_linear_new (dim);
+    t8_cmesh_register_geometry (cmesh, linear_geom);
+  }
+  t8_cmesh_set_partition_range (cmesh, 3, first_tree, last_tree);
+}
+
+/**
+ * Set the partition for cmesh coming from a distributed vtkGrid (like pvtu)
+ * 
+ * \param[in, out] cmesh On input a cmesh, on output a cmesh with a partition according to the number of trees read on each proc
+ * \param[in] mpirank The mpirank of this proc
+ * \param[in] mpisize The size of the mpicommunicator
+ * \param[in] num_trees The number of trees read on this proc. Must be >= 0
+ * \param[in] dim     The dimension of the cells
+ * \param[in] comm    The mpi-communicator to use. 
+ * \return            the global id of the first tree on this proc. 
+ */
+static t8_gloidx_t
+t8_vtk_distributed_partition (t8_cmesh_t cmesh, const int mpirank,
+                              const int mpisize,
+                              t8_gloidx_t num_trees, int dim,
+                              sc_MPI_Comm comm)
+{
+  t8_gloidx_t         first_tree;
+  t8_gloidx_t         last_tree;
+  int                 mpiret;
+  /* Compute the dimension of the cmesh. Is necessary, because procs can be empty. */
+  int                 dim_buf;
+  mpiret = sc_MPI_Allreduce (&dim, &dim_buf, 1, sc_MPI_INT, sc_MPI_MAX, comm);
+  SC_CHECK_MPI (mpiret);
+  t8_cmesh_set_dimension (cmesh, dim_buf);
+  t8_geometry_c      *linear_geom = t8_geometry_linear_new (dim_buf);
+  t8_cmesh_register_geometry (cmesh, linear_geom);
+
+  /* Compute the global id of the first tree on each proc. */
+  t8_shmem_init (comm);
+  t8_shmem_set_type (comm, T8_SHMEM_BEST_TYPE);
+  t8_shmem_array_t    offsets = NULL;
+  t8_shmem_array_init (&offsets, sizeof (t8_gloidx_t), mpisize + 1, comm);
+
+  t8_shmem_array_prefix ((void *) &num_trees, offsets, 1, T8_MPI_GLOIDX,
+                         sc_MPI_SUM, comm);
+
+  first_tree = t8_shmem_array_get_gloidx (offsets, mpirank);
+  /* Set the partition of the cmesh. */
+  if (num_trees == 0) {
+    last_tree = first_tree - 1;
+  }
+  else {
+    last_tree = first_tree + num_trees - 1;
+  }
+  t8_cmesh_set_partition_range (cmesh, 3, first_tree, last_tree);
```

Comment:
It might be good style to do the following. Self-documenting code.
```suggestion
  const int set_face_knowledge = 3 /* Expect face connection of local and ghost trees. */
  t8_cmesh_set_partition_range (cmesh, set_face_knowledge , first_tree, last_tree);
```